README.md

```diff
@@ -14,243 +14,222 @@
+- [Installation](#step-1-installing-of-component)
+- [Configuration](#step-2-configuration-of-component)
+- [Services, Events, Entities and Attributes](#step-3-checking-services-events-and-entities)
+- [Example automation](#step-4-creating-automation)
+
+### Step 1: Installing of Component
+Install the custom component (preferably using HACS) and then use the Configuration --> Integrations pane to search for 'Smart Irrigation'. In your sidebar you will find a new entry for panel 'Smart Irrigation'
+- API Key for Open Weather Map (optional). Only required in mode 1) and 3). See [Getting Open Weater Map API Key](#getting-open-weather-map-api-key) below for instructions.
+
+
+### Step 2: Configuration of Component
 You will need to specify the following:
+
+#### GENERAL
+This page provides global settings.
+- Automatic duration calculation: If enabled set the time of calculation (HH:MM).
+- Automatic weather data update: If enabled specify how often sensor update should happen (minutes, hours, days). Warning: weatherdata update time on or after calculation time! 
+
+#### ZONES
+Specify one or more irrigation zones here. The component calculates irrigation duration per zone, depending on size, throughput, state, module and mapping. A zone can be _disabled_ (so it doesn't do anything), _automatic_ or _manual_. If in automatic, duration is automatically calculated. If in manual, you specify the duration yourself. If disabled, the zone is not included in any calculations.
+
+> **When entering any values in the configuration of this component, keep in mind that the component will expect inches, sq ft, gallons, gallons per minute, or mm, m<sup>2</sup>, liters, liters per minute respectively depending on the settings in Home Assistant (imperial vs metric system).
+
+- Name: The name of your zone, e.g. garden
+- Size: The size of this zone (m<sup>2</sup> or sq ft)
+- Throughput: The flow of this zone (liter/minute or ???/minute)
+- Bucket: You can manipulate this calculated value.
+- Lead Time: In seconds. Time needed to warm up your irrigation system, e.g. time to establish a connection or start a pump etc.
+
+You can Update all zones and Callculate all.
+
+**Per Zone Settings**
+You can change any value mentioned before. Additionaly there are some more options.
+
+- State: Wether Updating and Calculation of that zone should be enabled in mode 'Automatic' or 'Manual' or 'Disabled'
+- Module: Choose the calculation module (see below) shall be used for that zone to calculate irrigation duration
+- Mapping: Which sensor mapping (see below) shall be used for that zone
+- Bucket: Either calculated or manualy set. Bucket >=0 no irrigation is necesarry, bucket <0 irrigation is necesarry.
+- Multiplier: Multiplies the duration of irrigation
+- Duration: Either calculated or manualy set
+
+Below each zone there are some buttons to update, calculate or delete that zone.
+
+#### MODULES
```

Comment:
also add the passthrough module!

Reply:
Never used., but found it
Passthrough module that returns the value of an Evapotranspiration sensor as delta.